### PR TITLE
mint vehicle

### PR DESCRIPTION
**Pull request by Allyson-English**

### CODE REVIEW

File: src/api/Method.ts [hunk 1]
```diff
@@ -1,147 +1,147 @@
...
+  /**
+   * Query Parameters
+   * Extract required queryParams from the resource object
+   */
+
+  const queryParams = resource.queryParams || {};
+  // Check if required queryParams match with incoming params
+  for (const key in queryParams) {
+    // We'll fail early if there's a missing required query param from the user
+    if (queryParams[key] === true && !params[key]) {
```
Comment (LorranSutter) [inline src/api/Method.ts:52]: Can we just do `queryParams[key] && !params[key]` ?

File: src/api/Method.ts [hunk 2]
```diff
@@ -1,147 +1,147 @@
...
-    let match;
-    while ((match = placeholderRegex.exec(resource.path)) !== null) {
-        const key = match[1];
-        const value = params[key];
-        if (value !== undefined) {
-            url = url.replace(match[0], value.toString());
-        }
+  const placeholderRegex = /:([\w]+)/g;
+  let match;
+  while ((match = placeholderRegex.exec(resource.path)) !== null) {
```
Comment (LorranSutter) [inline src/api/Method.ts:84]: I believe `exec` just returns either `null` or an array, but I think I would be more comfortable with `while (match = placeholderRegex.exec(resource.path))`

File: src/api/Method.ts [hunk 3]
```diff
@@ -1,147 +1,147 @@
...
+    const key = match[1];
+    const value = params[key];
+    if (value !== undefined) {
```
Comment (LorranSutter) [inline src/api/Method.ts:87]: I don't know if `value` could be `null` here, so I think it is safer to do just `if(value)`

File: src/dimo.test.ts [hunk 4]
```diff
@@ -1,76 +1,75 @@
...
+import { DIMO } from "./dimo";
 
-const PROD = 'Production';
-const DEV = 'Dev';
+const PROD = "Production";
+const DEV = "Dev";
 
-const dimo = new DIMO(PROD);
-const devDimo = new DIMO(DEV);
+// const dimo = new DIMO(PROD);
```
Comment (LorranSutter) [inline src/dimo.test.ts:6]: It doesn't look like these tests are going to run

File: src/dimo.ts [hunk 5]
```diff
@@ -1,110 +1,128 @@
...
+     */
+    this.web3 = new DimoWeb3Client(config);
+  }
+
+  // Helper Function
+  async authenticate() {
+    let fs;
+    try {
+      // Dynamically import fs
+      if (typeof process !== "undefined" && process.versions && process.versions.node) {
```
Comment (LorranSutter) [inline src/dimo.ts:84]: Can we just do `if (process && process.versions && process.versions.node)` ?

File: src/dimo.test.ts (hunk 4 above)
Comment (Allyson-English) [reply src/dimo.test.ts:6]: yeah this is all from what James had done previously.. commenting out for now!

**Pull request closed without merging**